doc/tutorials/tutorial_understanding_building_benchmark.rst

```diff
@@ -0,0 +1,370 @@
+
+    git clone git@github.com:benchopt/template_benchmark my_benchmark
+
+The template benchmark is not designed to model our ridge regression problem, but luckily it is pretty close!
+The cost which is implemented in the template benchmark is the Ordinary Least Squares (OLS)
+
+.. math:: 
+
+    g(\beta) = \frac{1}{2} \|y - X\beta \|^2
+
+and the solver implement is GD with \\(\\nabla g(\\beta) = -X^Ty + X^TX\\beta \\) the gradient of \\(g\\) at \\(\\beta \\).
+
+We will modify this template to adapt it to the ridge regression problem next, but before that let us run this benchmark.
+In other words, let us use benchopt to read the contents of the template benchmark and run GD on OLS with a predefined set of stepsizes, in our case \\( [1, 1.99] \\) (the stepsize is scaled by the inverse of the gradient's Lipschitz constant, you can ignore this detail if you are not familiar with this concept).
+
+To run the template benchmark, simply run the following command in the terminal:
+
+.. prompt:: bash $
+
+    benchopt run my_benchmark
+
+
+.. image:: console_template.png
+    :width: 600
+    :alt: A screenshot of the console during the benchopt run command
+
+You will see something similar to this in your terminal
+
+    - *Simulated* tells us that the dataset run by benchopt is the simulation set up in the benchmark
+    - *Ordinary Least Squares* tells us which loss is minimized, and the hyperparameters are written in bracket. 
+    - *GD* is a line indicating the progress of algorithm GD for this problem (Simulated dataset, OLS loss). Again its hyperparameters are written in brackets (here the stepsize value).
+
+Once the benchmark has been run, a window should open in your default navigator.
+This provides a visualization of the results of the run, which is useful to immediately observe, comment and share the results.
+After running the template benchmark, we can observe the convergence plots of GD with the two different stepsize choices, for two different simulated dataset.
+The convergence plots can be made log-log for easier reading.
+Also, depending on your computer, the runtime is so low that the convergence plot with respect to time is not reliable.
+We advise to look at the results in terms of iteration (scrolling menu on the bottom left of the webpage).
+Feel free to play around with the plotting options here!
+Note that the dataset comes with two values for an option whiten, let us ignore this detail in the tutorial.
+
+Exploring the benchmark structure
+---------------------------------
+
+The template benchmark we are using at the moment is not exactly encoding the information we need for the ridge regression problem.
+To properly modify the benchmark, first we need to dive deeper into how benchmarks work.
+To follow through this section, it is advised to open an editor (like vscode) in the root of the template benchmark to easily navigate between the files and folders.
+
+Here is the architecture of our template benchmark:
```

Comment:
I'm in favor of not displaying the files that we don't speak about : clean template, test config, gitignore. It makes it easier for the reader to focus on the key concepts.

Reply:
Ok sure